.ci/pitest-survival-check-xml.groovy

```diff
@@ -205,7 +206,7 @@ private static Mutation getMutation(Node mutationNode) {
         final String javaExtension = ".java"
         final String mutationFilePath = startingPath + mutationFileName
                 .substring(0, mutationFileName.length() - javaExtension.length())
-                .replaceAll("\\.", SEPARATOR) + javaExtension
+                .replace(".", SEPARATOR) + javaExtension
```

Comment:
@Vyom-Yadav why did we change replaceAll to replace? Don't we have multiple separators in the path?

Reply:
This is the reason I used `replaceAll(..)` initially, but apparently, both methods technically replace all. From Javadoc of `replace(..)`:
```java
/**                                                                       
 * Replaces each substring of this string that matches the literal target 
 * sequence with the specified literal replacement sequence. The          
 * replacement proceeds from the beginning of the string to the end, for  
 * example, replacing "aa" with "b" in the string "aaa" will result in    
 * "ba" rather than "ab".                                                 
 *                                                                        
 * @param  target The sequence of char values to be replaced              
 * @param  replacement The replacement sequence of char values            
 * @return  The resulting string                                          
 * @since 1.5                                                             
 */                                                                       
```

Reply:
In Java, the only difference between `replace` and `replaceAll` is the parameter it takes. One is a regular expression and the other takes a normal string. If you are not using regular expressions, then you should just use `replace`.